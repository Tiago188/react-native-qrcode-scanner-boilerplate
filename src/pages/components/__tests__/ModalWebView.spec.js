import React from 'react';
import { shallow } from 'enzyme';

import ModalWebView from '../ModalWebView';

const props = {

    modalVisible: true,
    url: 'https://google.com.br',
    openLink: jest.fn(),
    handleButton: jest.fn()

};

it('The modal is closed', () => {
    const wrapper = shallow(<ModalWebView {...props} modalVisible={false} />)
    
    expect(wrapper).toMatchSnapshot()
});

it('renders correctly', () => {
    const wrapper = shallow(<ModalWebView {...props} />)
    
    expect(wrapper).toMatchSnapshot()
});