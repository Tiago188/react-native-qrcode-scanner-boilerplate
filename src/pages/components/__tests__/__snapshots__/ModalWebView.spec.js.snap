// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The modal is closed 1`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 1264,
          "width": 700,
        }
      }
    >
      <WebView
        automaticallyAdjustContentInsets={false}
        domStorageEnabled={true}
        javaScriptEnabled={true}
        originWhitelist={
          Array [
            "http://*",
            "https://*",
          ]
        }
        source={
          Object {
            "uri": "https://google.com.br",
          }
        }
        startInLoadingState={true}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-around",
          "paddingTop": 10,
          "width": 700,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <Text>
          Abrir no navegador
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <Text>
          Ler outro QRCode
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</Component>
`;

exports[`renders correctly 1`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 1264,
          "width": 700,
        }
      }
    >
      <WebView
        automaticallyAdjustContentInsets={false}
        domStorageEnabled={true}
        javaScriptEnabled={true}
        originWhitelist={
          Array [
            "http://*",
            "https://*",
          ]
        }
        source={
          Object {
            "uri": "https://google.com.br",
          }
        }
        startInLoadingState={true}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-around",
          "paddingTop": 10,
          "width": 700,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <Text>
          Abrir no navegador
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <Text>
          Ler outro QRCode
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</Component>
`;
